### PR TITLE
Update Existing Go Templates to use Go 1.11.5

### PR DESCRIPTION
**Pull request by ernestojeda**
Changes made:
  - updated `shell/install_custom_golang.sh` to install Go version 1.11.5
  - change `GOROOT` to be `/opt/go-custom/go` to remove any version numbers in the path. This way future updates should not affect the jjb templates only the install script.
  - updated templates referencing the `shell/install_custom_golang.sh` to install only on the master stream

Signed-off-by: Ernesto Ojeda <ernesto.ojeda@intel.com>

### CODE REVIEW

**Comment by JPWKU** [comment]
This will let us still use the current go version for delhi release correct?

**Comment by ernestojeda** [comment]
> This will let us still use the current go version for delhi release correct?

Sorry, what is meant by current version. Do you mean the current version installed on the `centos7-docker-4c-2g` build node? If so, then dehli would still continue to use whatever version of Go is installed on that node which I think is Go 1.9

**Comment by ernestojeda** [comment]
There will be corresponding Dockerfile changes in the following repos as well:
  - edgex-go (https://github.com/edgexfoundry/edgex-go/pull/1126)
  - device-sdk-go (https://github.com/edgexfoundry/device-sdk-go/pull/177)
  - device-modbus-go (https://github.com/edgexfoundry/device-modbus-go/pull/20)

Sandbox Jobs building all the PR's:
  - edgex-go (https://jenkins.edgexfoundry.org/sandbox/job/edgex-go-master-verify-go/4/console) commit hash `bf29cfadfa04e81d0f24cc2cd22f4c582e3e6353`
  - device-sdk-go (https://jenkins.edgexfoundry.org/sandbox/job/device-sdk-go-master-verify-go/2/console) commit hash `e4010a75c68b9fa7badfc258804f0b9847857b6e`
  - device-modbus-go (https://jenkins.edgexfoundry.org/sandbox/job/device-modbus-go-master-verify-go/5/console) commit hash `cbe1557637429110e272b948ea0c6e2a05afaffc`

**Review by eb-oss — APPROVED** [review]
This looks much better to me.
I'm going to await one more approval before merging.

**Review by jamesrgregg — APPROVED** [review]
LGTM

File: jjb/device/device-sdk-go.yaml
Comment (dweomer) [inline jjb/device/device-sdk-go.yaml:15]: `make test` is invoked twice on master? (see `pre_build_script` above)

also, it doesn't make sense that the `test` target is being invoked before the `build` target

**Review by dweomer — COMMENTED** [review]
please fix `make` target invocation order

File: jjb/device/device-sdk-go.yaml
Comment (dweomer) [reply jjb/device/device-sdk-go.yaml:15]: too late!

**Pull request merged**
